packages/SwingSet/src/vats/timer/vat-timer.js

```diff
@@ -1,98 +1,854 @@
 // @ts-check
 
-import { Nat } from '@agoric/nat';
-import { assert, details as X } from '@agoric/assert';
+import { E } from '@endo/eventual-send';
 import { Far, passStyleOf } from '@endo/marshal';
-import { makeNotifierFromAsyncIterable } from '@agoric/notifier';
 import { makePromiseKit } from '@endo/promise-kit';
-
-import { makeTimedIterable } from './timed-iteration.js';
+import { Nat } from '@agoric/nat';
+import { assert } from '@agoric/assert';
+import {
+  provideKindHandle,
+  defineDurableKind,
+  defineDurableKindMulti,
+  makeScalarBigMapStore,
+  makeScalarBigWeakMapStore,
+} from '@agoric/vat-data';
+import { provide, makeLegacyWeakMap } from '@agoric/store';
 import { TimeMath } from './timeMath.js';
 
-export function buildRootObject(vatPowers) {
+// RAM usage: O(number of outstanding delay() promises) +
+// O(number of notifiers??). setWakeup() only consumes DB.
+
+/**
+ * // Handler is a Far object with .wake(time) used for callbacks
+ *
+ * @typedef Handler
```

Comment:
Whether this is an improvement depends on what you intend. By putting it immediately after the `@typedef Handler`, when some elsewhere hovers over the `Handler` type, they'll see your comment. OTOH, if you intend this comment only for those reading the implementation, no change suggested.

```suggestion
 * @typedef Handler
 * Handler is a Far object with .wake(time) used for callbacks
```

Reply:
Actually, if you're going to leave it as implementation comment, please drop the extraneous `//` within the `/** ... */` comment